django_opentracing/tracing.py

```diff
@@ -134,8 +135,23 @@ def _finish_tracing(self, request, response=None, error=None):
             scope.span.set_tag("sfx.error.object", str(error.__class__))
             scope.span.set_tag("sfx.error.kind", error.__class__.__name__)
             scope.span.set_tag("sfx.error.stack", traceback.format_exc())
+
         if response is not None:
             scope.span.set_tag(tags.HTTP_STATUS_CODE, response.status_code)
+            if self._trace_response_header_enabled:
+                trace_id = getattr(scope.span.context, "trace_id", 0)
+                span_id = getattr(scope.span.context, "span_id", 0)
```

Comment:
I think that does it does NOT make any sense to add the headers if `trace_id` or  `span_id` are `0`

At least in Go instrumentation, we are not adding these headers for such case:
- https://github.com/signalfx/splunk-otel-go/blob/main/instrumentation/net/http/splunkhttp/server_timing.go#L30
- https://github.com/signalfx/signalfx-go-tracing/blob/master/contrib/internal/httputil/trace.go#L47

probably it would be good to consider extracting to another function/middleware/decorator as the `if` nesting would get little nasty

please double check this in other Python instrumentation as I might have not spotted it before

Reply:
Makes sense. I did  not have the 0 checks at all in first place but some tests in tornado used a noop tracer that resulted in trace_id attribute not being present at all and I added the defensive checks as a result without thinking it through :)

Thanks for the suggested. Will update it.